core/services/ocr3/plugins/ccip/internal/reader/home_chain.go

```diff
@@ -0,0 +1,285 @@
+	lggr            logger.Logger
+	mutex           *sync.RWMutex
+	state           state
+	failedPolls     uint
+	// How frequently the poller fetches the chain configs
+	pollingDuration time.Duration
+}
+
+const MaxFailedPolls = 10
+
+func NewHomeChainConfigPoller(
+	homeChainReader types.ContractReader,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) *HomeChainConfigPoller {
+	return &HomeChainConfigPoller{
+		stopCh:          make(chan struct{}),
+		homeChainReader: homeChainReader,
+		state:           state{},
+		mutex:           &sync.RWMutex{},
+		failedPolls:     0,
+		lggr:            lggr,
+		pollingDuration: pollingInterval,
+	}
+}
+
+func (r *HomeChainConfigPoller) Start(ctx context.Context) error {
+	err := r.fetchAndSetConfigs(ctx)
+	if err != nil {
+		// Just log, don't return error as we want to keep polling
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+	}
+	r.lggr.Infow("Start Polling ChainConfig")
+	return r.sync.StartOnce(r.Name(), func() error {
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *HomeChainConfigPoller) poll() {
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	ticker := time.NewTicker(r.pollingDuration)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			r.mutex.Lock()
+			r.failedPolls = 0
+			r.mutex.Unlock()
+			return
+		case <-ticker.C:
+			if err := r.fetchAndSetConfigs(ctx); err != nil {
+				r.mutex.Lock()
+				r.failedPolls++
+				r.mutex.Unlock()
+				r.lggr.Errorw("Fetching and setting configs failed", "err", err)
+			}
+		}
+	}
+}
+
+func (r *HomeChainConfigPoller) fetchAndSetConfigs(ctx context.Context) error {
+	chainConfigInfos, err := r.fetchOnChainConfig(ctx)
+	if err != nil {
+		r.lggr.Errorw("Fetching on-chain configs failed", "err", err)
+		return err
+	}
+	if len(chainConfigInfos) == 0 {
+		// That's a legitimate case if there are no chain configs on chain yet
+		r.lggr.Warnw("no on chain configs found")
+		return nil
+	}
+	homeChainConfigs, err := r.convertOnChainConfigToHomeChainConfig(chainConfigInfos)
+	if err != nil {
+		r.lggr.Errorw("error converting OnChainConfigs to ChainConfig", "err", err)
+		return err
+	}
+	r.lggr.Infow("Setting ChainConfig")
+	r.setState(homeChainConfigs)
+	return nil
+}
+
+func (r *HomeChainConfigPoller) setState(chainConfigs map[cciptypes.ChainSelector]ChainConfig) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	s := &r.state
+	s.chainConfigs = chainConfigs
+	s.nodeSupportedChains = createNodesSupportedChains(chainConfigs)
+	s.knownSourceChains = createKnownChains(chainConfigs)
+	s.fChain = createFChain(chainConfigs)
+}
+
+func (r *HomeChainConfigPoller) GetChainConfig(chainSelector cciptypes.ChainSelector) (ChainConfig, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	s := r.state
+	if _, ok := s.chainConfigs[chainSelector]; !ok {
+		return ChainConfig{}, fmt.Errorf("chain config not found for chain %v", chainSelector)
+	}
+	return s.chainConfigs[chainSelector], nil
```

Comment:
nit: you can do this with a single lookup
```suggestion
	if chainConfig, ok := s.chainConfigs[chainSelector]; ok {
		return chainConfig, nil
	}
	return ChainConfig{}, fmt.Errorf("chain config not found for chain %v", chainSelector)
```